test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -279,6 +279,54 @@ struct test_transform_iterator {
     }
 };
 
+struct test_permutation_iterator
+{
+    template <typename T1, typename T2>
+    void
+    operator()(::std::vector<T1>& in1, ::std::vector<T2>& in2)
+    {
+        T1 iota_max = ::std::numeric_limits<T1>::max() < in1.size() ? ::std::numeric_limits<T1>::max() : in1.size();
+        ::std::iota(in1.begin(), in1.begin() + iota_max, T1(0));
+        ::std::reverse_copy(in1.begin(), in1.begin() + iota_max, in2.begin());
+
+        oneapi::dpl::permutation_iterator<typename ::std::vector<T1>::iterator, typename ::std::vector<T2>::iterator> perm_begin;
+        perm_begin = oneapi::dpl::make_permutation_iterator(in1.begin(), in2.begin());
+        auto perm_end = oneapi::dpl::make_permutation_iterator(in1.begin(), in2.begin()) + iota_max;
+
+        ::std::vector<T1> result(iota_max);
+        ::std::copy(perm_begin, perm_end, result.begin());
+
+        EXPECT_TRUE(::std::is_sorted(result.begin(), result.end(), ::std::greater<T1>()),
+                    "wrong result from permutation_iterator");
+
+        oneapi::dpl::permutation_iterator<typename ::std::vector<T1>::iterator, typename ::std::vector<T2>::iterator> perm_it1(in1.begin(), in2.begin());
+        oneapi::dpl::permutation_iterator<typename ::std::vector<T1>::iterator, typename ::std::vector<T2>::iterator> perm_it2(in1.begin(), in2.begin(), in2.size()-1);
+        EXPECT_TRUE(perm_it1 == perm_begin, "wrong result from permutation_iterator");
+        EXPECT_TRUE(perm_it2 == perm_begin + in2.size()-1, "wrong result from permutation_iterator");
+        EXPECT_TRUE(perm_it1.base() == in1.begin(), "wrong result from permutation_iterator");
+        EXPECT_TRUE(perm_it1.map() == in2.begin(), "wrong result from permutation_iterator");
```

Comment:
I'd suggest to change output strings depending on an expected behavior, e.g. "wrong result from map()", "wrong result from base()"

Reply:
I've updated them so they're similar to output strings in `transform_iterator`.